docs/io-app/load-next-page-message-list.puml

```diff
@@ -0,0 +1,70 @@
+@startuml name LoadNextMessagePage-ioApp
+participant MessageList
+participant homeUtils
+database Redux
+participant allPaginatedReducer
+participant handleLoadNextPageMessages
+rnote over handleLoadNextPageMessages
+ saga
+endrnote
+entity BE
+
+activate MessageList
+rnote over MessageList
+User scrolls to
+the bottom of
+the message list
+endrnote
+MessageList -> MessageList : FlatList invokes\nonEndReachedCallback
+activate MessageList
+
+MessageList -> homeUtils : calls\ngetLoadNextPageMessagesActionIfAllowed
+activate homeUtils
+rnote over homeUtils
+undefined is returned if
+- system is busy archiving or
+  restoring messages
+- system is loading the inbox
+  or archive message list
+- there is no 'next' id for
+  current message page
+- a previous request for next
+  page messages has failed
+  within two seconds
+endrnote
+homeUtils -> MessageList : returns\n| undefined\n| loadNextPageMessages.request
+deactivate homeUtils
+
+MessageList -> Redux : dispatches\nloadNextPageMessages.request
+deactivate MessageList
+deactivate MessageList
+
+Redux -> allPaginatedReducer ++ : passes\nloadNextPageMessages.request
+allPaginatedReducer -> Redux --: returns\nnew AllPaginated state
+
+Redux -> handleLoadNextPageMessages ++ : forwards\nloadNextPageMessages.request
+
+handleLoadNextPageMessages  -> BE : API call to\nmessages?maximum_id=
+rnote over BE
+async
+endrnote
+BE ->handleLoadNextPageMessages : replies with\nPaginatedPublicMessagesCollection
```

Comment:
You can use a dotted arrow instead of a note to highlight that the operation is asynchronous

Reply:
Done it!